.github/workflows/ci-test.yml

```diff
@@ -255,7 +268,7 @@ jobs:
     name: Run land blocking test
     runs-on: self-hosted
     needs: [prepare, build-images, need-base-images]
-    if: ${{ always() && needs.build-images.result=='success' }}
+    if: ${{ github.event_name == 'push' && needs.prepare.outputs.test-land-blocking == 'true' && needs.build-images.result=='success' }}
```

Comment:
I think we need to keep always() here, one scenario is: if we do not need base image, and that step will be skipped, then this job will be skipped as well since  GH always check needs first

Reply:
it just feels weird we need to logically "and" always() with another condition for this to work in GHA.

Reply:
yea, but as now we have 2 flags, one for enable/disable LBT and another for dispatch lbt_comp and lbt suite. So we actually have 3 routines: 1. enable LBT run with lbt_comp, 2. enable LBT run with lbt. 3. disable LBT.

for this case if we do not have always here, the case 2 will never happen then

Reply:
we can do a canary test with:
KILL_SWITCH_LAND_BLOCKING_TEST = true
KILL_SWITCH_LAND_BLOCKING_COMPAT = false
I assume lbt test will be skip

Reply:
done